spec/lib/good_job/job_spec.rb

```diff
@@ -265,10 +265,25 @@ def perform(result_value = nil, raise_error: false)
         expect(error).to be_a(ExpectedError)
       end
 
+      describe 'GoodJob.reperform_jobs_on_standard_error' do
+        it 'is deprecated' do
+          current_preserve_job_records = GoodJob.preserve_job_records
+          allow(ActiveSupport::Deprecation).to receive(:warn)
+
+          GoodJob.reperform_jobs_on_standard_error = false
+
+          expect(GoodJob.preserve_job_records).to be(false)
+          expect(GoodJob.reperform_jobs_on_standard_error).to be(false)
+          expect(ActiveSupport::Deprecation).to have_received(:warn).twice
+        ensure
+          GoodJob.preserve_job_records = current_preserve_job_records
+        end
+      end
```

Comment:
It would be best for this to live in in `good_job_spec.rb` instead of `job_spec.rb` because these are testing methods on `GoodJob`.

Here is some suggested (potentially overly-) idiomatic RSpec to replace the `ensure` reset with an `around` block, and a compound expectation that assigning a value changes the values:

```suggestion
  describe '.reperform_jobs_on_standard_error' do
    around do |example|
      original_retry_on_unhandled_error = GoodJob.retry_on_unhandled_error
      example.run
      GoodJob.retry_on_unhandled_error = original_retry_on_unhandled_error
    end

    it 'is deprecated and replaced with .retry_on_unhandled_error' do
      allow(ActiveSupport::Deprecation).to receive(:warn)
      GoodJob.retry_on_unhandled_error = true

      expect {
        GoodJob.reperform_jobs_on_standard_error = false
      }.to(
        change(GoodJob, :retry_on_unhandled_error).from(true).to(false)
          .and change(GoodJob, :reperform_jobs_on_standard_error).from(true).to(false)
      )

      expect(ActiveSupport::Deprecation).to have_received(:warn).exactly(3).times
    end
  end
```

Reply:
done